src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -225,21 +238,17 @@ public boolean executeTask(
     Optional<PantsOptions> pantsOptional = PantsOptions.getPantsOptions(currentProject);
     if (!pantsOptional.isPresent()) {
       showPantsMakeTaskMessage("Pants Options not found.\n", ConsoleViewContentType.ERROR_OUTPUT, currentProject);
-      return Pair.create(false, Optional.empty());
+      return PantsExecuteTaskResult.emptyFailure();
     }
-
     PantsOptions pantsOptions = pantsOptional.get();
 
     /* Global options section. */
     commandLine.addParameter(PantsConstants.PANTS_CLI_OPTION_NO_COLORS);
 
-    if (useCleanAll) {
-      commandLine.addParameter("clean-all");
-      if (pantsOptions.supportsAsyncCleanAll()) {
-        commandLine.addParameter(PantsConstants.PANTS_CLI_OPTION_ASYNC_CLEAN_ALL);
-      }
+    if (tasks.contains(PantsConstants.PANTS_TASK_CLEAN_ALL) && pantsOptions.supportsAsyncCleanAll()) {
+      commandLine.addParameter(PantsConstants.PANTS_CLI_OPTION_ASYNC_CLEAN_ALL);
     }
-    if (pantsOptions.supportsManifestJar()) {
+    if (tasks.contains(PantsConstants.PANTS_TASK_EXPORT_CLASSPATH) && pantsOptions.supportsManifestJar()) {
```

Comment:
the oldest pants version we are validating is 1.2.0, and it has the option, so we can safely remove this check now.

Reply:
removed, thanks for the explanation